data/blog/enabling-out-of-the-box-performance-insights-in-the-unity-sdk.md

```diff
@@ -0,0 +1,188 @@
+/// </summary>
+public partial class SentryMonoBehaviour
+{
+    public void StartAwakeSpan(MonoBehaviour monoBehaviour) =>
+        SentrySdk.GetSpan()?.StartChild("awake", $"{monoBehaviour.gameObject.name}.{monoBehaviour.GetType().Name}");
+
+    public void FinishAwakeSpan() => SentrySdk.GetSpan()?.Finish(SpanStatus.Ok);
+}
+```
+
+Initially, we did this change manually and took a look at the resulting IL.
+
+```csharp
+private void Awake()
+{
+    SentryMonoBehaviour.Instance.StartAwakeSpan(this);
+
+    Debug.Log("Hello World!");
+
+    SentryMonoBehaviour.Instance.FinishAwakeSpan();
+}
+```
+
+```csharp
+	// Methods
+	.method private hidebysig
+		instance void Awake () cil managed
+	{
+		// Method begins at RVA 0x22f7
+		// Header size: 1
+		// Code size: 32 (0x20)
+		.maxstack 8
+
+		// SentryMonoBehaviour.Instance.StartAwakeSpan(this);
+		IL_0000: call class [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour::get_Instance()
+		IL_0005: ldarg.0
+		IL_0006: callvirt instance void [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour::StartAwakeSpan(class [UnityEngine]UnityEngine.MonoBehaviour)
+		// Debug.Log("Hello World!");
+		IL_000b: ldstr "Hello World!"
+		IL_0010: call void [UnityEngine.CoreModule]UnityEngine.Debug::Log(object)
+		// SentryMonoBehaviour.Instance.FinishAwakeSpan();
+		IL_0015: call class [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour::get_Instance()
+		IL_001a: call instance void [Sentry.Unity]Sentry.Unity.SentryMonoBehaviour::FinishAwakeSpan()
+		// }
+		IL_001f: ret
+	} // end of method BlogMaterial::Awake
+```
+
+But the `.dll` is not just a text file. So how do we modify this? Luckily, there are libraries like [Cecil](https://github.com/jbevain/cecil) around that we can build on and that do the heavy lifting for us. Cecil basically turns this into something akin to painting by numbers:
+
+1. Compile your code to create the “baseline”
+2. Modify the source code to your desired end-result
+3. Let the compiler do what it does best - translate your C# code into IL
+4. Inspect the difference between the baseline and the end-result
+5. Use Cecil to recreate the change
+
+The code that modifies the `Awake` and adds the `StartSpan` functionality is three lines:
+
+```csharp
+// Adding in reverse order because we're inserting *before* the 0ths element
+processor.InsertBefore(method.Body.Instructions[0], processor.Create(OpCodes.Callvirt, startAwakeSpanMethod));
+processor.InsertBefore(method.Body.Instructions[0], processor.Create(OpCodes.Ldarg_0));
+processor.InsertBefore(method.Body.Instructions[0], processor.Create(OpCodes.Call, getInstanceMethod));
+```
+
+You can inspect the whole setup of reading, modifying and writing the IL [here](https://github.com/getsentry/sentry-unity/blob/main/src/Sentry.Unity.Editor/AutoInstrumentation/SentryPerformanceAutoInstrumentation.cs).
```

Comment:
use a commit sha instead of `main` so the lin doesn't break if we refactor/move stuff around